pages/dropdown/width.page.tsx

```diff
@@ -207,12 +212,17 @@ function CustomSelect({ expandToViewport, loading, onOpen, onClose, virtualScrol
 
 export default function () {
   const { urlParams } = useContext(AppContext as DemoContext);
-  const { asyncLoading, component, triggerWidth, virtualScroll, expandToViewport, containerWidth } = urlParams;
+  const { asyncLoading, component, triggerWidth, virtualScroll, expandToViewport, containerWidth, manualServerMock } =
+    urlParams;
   const [loading, setLoading] = useState(asyncLoading);
   const onOpen = () => {
     if (asyncLoading) {
       setLoading(true);
-      setTimeout(() => setLoading(false), 500);
+      if (manualServerMock) {
+        window.__pendingCallbacks.push(() => setLoading(false));
```

Comment:
wouldn't it be easier to configure a synchronous response or to create an additional page with synchronous data flow instead of introducing these race condition handling?

Reply:
This page seems quite complex and I think it is likely that copying it would result in a lot of duplicated code. The manualServerMock only replaces the timeout with an explicit command to simplify integ tests and eliminate flakiness.